src/core/lib/event_engine/event_engine_factory.cc

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <grpc/support/port_platform.h>
+
+#include <grpc/event_engine/event_engine.h>
+
+namespace grpc_event_engine {
+namespace experimental {
+
+EventEngine* g_event_engine = nullptr;
+std::function<std::unique_ptr<EventEngine>()>* g_event_engine_factory = nullptr;
+
+EventEngine* GetDefaultEventEngine() {
```

Comment:
It seems like it shouldn't be safe to have no synchronization here?

(what if two threads discover no event engine simultaneously)

Reply:
Yes, good catch. I've add a lock around factory access & mutations, and made the global EventEngine instance a function-scoped static variable.

I was thinking about our immediate use cases, where we expect a custom default factory to be set once, during gRPC one-time initialization. But that's still not a good reason to forget to do synchronization here.